### PR TITLE
implements orderService, userService, shoppingCartService, and Dao la…

### PR DESCRIPTION
**Pull request by vladkuzya**
…yer for them

### CODE REVIEW

File: src/main/java/mate/academy/dao/impl/OrderDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import mate.academy.db.Storage;
+import mate.academy.lib.Dao;
+import mate.academy.model.Order;
+import mate.academy.model.ShoppingCart;
+
+@Dao
+public class OrderDaoImpl implements OrderDao {
+    @Override
+    public Order create(ShoppingCart shoppingCart) {
+        Order order = new Order(shoppingCart.getUserId());
```
Comment (TimoxaV) [inline src/main/java/mate/academy/dao/impl/OrderDaoImpl.java]: I guess it is better to create new Order in Service. In DAO you should only add it in DB.

File: src/main/java/mate/academy/dao/impl/UserDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    public List<User> getAll() {
+        return Storage.users;
+    }
+
+    @Override
+    public User update(User user) {
+        IntStream.range(0, Storage.users.size())
+                .filter(i -> Storage.users.get(i).getId().equals(user.getId()))
+                .forEach(i -> Storage.users.set(i, user));
+        return null;
```
Comment (TimoxaV) [inline src/main/java/mate/academy/dao/impl/UserDaoImpl.java]: You should return updated user. Null is bad for debug.

Comment (vladkuzya) [reply src/main/java/mate/academy/dao/impl/UserDaoImpl.java]: done)

File: src/main/java/mate/academy/dao/impl/OrderDaoImpl.java (hunk 1 above)
Comment (vladkuzya) [reply src/main/java/mate/academy/dao/impl/OrderDaoImpl.java]: done

File: src/main/java/mate/academy/db/Storage.java [hunk 3]
```diff
@@ -2,15 +2,42 @@
...
+
+    public static void addUser(User user) {
+        userId++;
+        user.setId(userId);
+        users.add(user);
+    }
+
+    public static void addShoppingCarts(ShoppingCart shoppingCart) {
+        shoppingCartId++;
+        shoppingCart.setId(shoppingCartId);
```
Comment (RomanDubovskyi) [inline src/main/java/mate/academy/db/Storage.java]:
```suggestion
        shoppingCart.setId(++shoppingCartId);
```
same in other methods

File: src/main/java/mate/academy/service/ShoppingCartService.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import mate.academy.model.ShoppingCart;
+
+public interface ShoppingCartService {
+    ShoppingCart create(ShoppingCart shoppingCart);
+
+    ShoppingCart addProduct(ShoppingCart shoppingCart, Product product);
+
+    boolean deleteProduct(ShoppingCart shoppingCart, Product product);
+
+    void clear(ShoppingCart shoppingCart); //remove all products from the shoppingCart
```
Comment (RomanDubovskyi) [inline src/main/java/mate/academy/service/ShoppingCartService.java]: we can remove comment

File: src/main/java/mate/academy/service/impl/OrderServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    private OrderDao orderDao;
+
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        Order order = new Order(shoppingCart.getUserId());
+        order.setProducts(new ArrayList<>(shoppingCart.getProducts()));
+        shoppingCart.getProducts().removeAll(shoppingCart.getProducts());
```
Comment (RomanDubovskyi) [inline src/main/java/mate/academy/service/impl/OrderServiceImpl.java]: use shoppingCartService.clear() instead, cause now you don't update shoppings cart from storage after you changed products field

File: src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    private ProductDao productDao;
+
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingCart) {
+        return shoppingCartDao.create(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart addProduct(ShoppingCart shoppingCart, Product product) {
+        shoppingCart.getProducts().add(productDao.getById(product.getId()).get());
```
Comment (RomanDubovskyi) [inline src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java]:
```suggestion
        shoppingCart.getProducts().add(product);
        return shoppingCartDao.update(shoppingCart);
```

File: src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        return shoppingCart;
+    }
+
+    @Override
+    public boolean deleteProduct(ShoppingCart shoppingCart, Product product) {
+        return shoppingCart.getProducts().remove(product);
```
Comment (RomanDubovskyi) [inline src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java]: same here, don't forget to update your cart after editing

File: src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    @Override
+    public void clear(ShoppingCart shoppingCart) {
```
Comment (RomanDubovskyi) [inline src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java:36]: same here

File: src/main/java/mate/academy/db/Storage.java (hunk 3 above)
Comment (vladkuzya) [reply src/main/java/mate/academy/db/Storage.java]: done)

File: src/main/java/mate/academy/service/ShoppingCartService.java (hunk 4 above)
Comment (vladkuzya) [reply src/main/java/mate/academy/service/ShoppingCartService.java]: done)

File: src/main/java/mate/academy/service/impl/OrderServiceImpl.java (hunk 5 above)
Comment (vladkuzya) [reply src/main/java/mate/academy/service/impl/OrderServiceImpl.java]: done)

File: src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java (hunk 6 above)
Comment (vladkuzya) [reply src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java]: done)

File: src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java (hunk 7 above)
Comment (vladkuzya) [reply src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java]: done)

File: src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java (hunk 8 above)
Comment (vladkuzya) [reply src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java:36]: done)

File: src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+        shoppingCart.getProducts().clear();
+        shoppingCartDao.update(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart getByUserId(Long userId) {
+        return shoppingCartDao.getById(userId).get();
```
Comment (RomanDubovskyi) [inline src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java]: you're just using here getById(Long cartId), but you're passing userId. 
Create separate method getByUserId(Long userId); on dao layer

File: src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+import mate.academy.model.ShoppingCart;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Inject
+    private ProductDao productDao;
```
Comment (RomanDubovskyi) [inline src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java]: we don't use this variable anywhere in this class

File: src/main/java/mate/academy/service/impl/OrderServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+    @Inject
+    private OrderDao orderDao;
+
+    @Inject
+    private ShoppingCartService shoppingCartService;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        Order order = new Order(shoppingCart.getUserId());
+        order.setProducts(new ArrayList<>(shoppingCart.getProducts()));
```
Comment (RomanDubovskyi) [inline src/main/java/mate/academy/service/impl/OrderServiceImpl.java]:
```suggestion
        order.setProducts(List.copyOf(shoppingCart.getProducts())));
```
https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_14_2.html

File: src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java (hunk 9 above)
Comment (vladkuzya) [reply src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java]: done)

File: src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java (hunk 10 above)
Comment (vladkuzya) [reply src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java]: done)

File: src/main/java/mate/academy/service/impl/OrderServiceImpl.java (hunk 11 above)
Comment (vladkuzya) [reply src/main/java/mate/academy/service/impl/OrderServiceImpl.java]: done)

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**